doc/release/1.10.0-notes.rst

```diff
@@ -50,6 +50,13 @@ See below for more details on these changes.
 Compatibility notes
 ===================
 
+Default casting rule change
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Default casting for inplace operations has changed to ``'same_kind'``. For
+instance, floats will no longer be automatically cast to ints. This will
+break code that has not been fixed to eliminate the DeprecationWarning
+raised since Numpy 1.7.
```

Comment:
I think an example would help? Maybe:

Default casting for inplace operations has changed to `'same_kind'`. For instance, if `n` is an array of integers, and `f` is an array of floats, then in previous versions of NumPy, an operation like `n += f` would silently cast the float array to integer (discarding the fractional part) so as to perform the operation in-place. Now, this operation will raise an exception. (If you want to upcast the out array to float, then use `n = n + f`). This will break code that has not been fixed to eliminate the DeprecationWarning added in NumPy 1.7.